RocketChat/Rocket.Chat.ReactNative
# fix: commonmark code_block

## Proposed changes


I needed to create a PR at the repo RocketChat/commonmark.js - https://github.com/RocketChat/commonmark.js/pull/5

> It's fixing the regex to close the code blocks at these lines: https://github.com/RocketChat/commonmark.js/blob/bb03ae0b380152a3286f910fddc9d4d66a978ad4/dist/commonmark.js#L9417-L9426
>
>Before, the regex was catching only if you did a single white space or one break line, if you added more then one break line at the end of the three backtick, the regex wasn't catching it and the block was never "closing". Also, I added the `m` to catch more then one code block in a single message, otherwise the regex was wrong.

## Issue(s)	


## How to test or reproduce


- Use a Server that you are able to disable the message parser Servers less than 6.5, then disable the message parser
- send to a channel 
```
 -```
 -Code block 1
 -```
 
 
 -New code block
 
 -```
 -Code block 2
 -```
```


- If you don't have access to a server less than 6.4 you can test send the message above in any server, and the FIRST render of the message you will be able to see this difference, for example:

#### Before

https://github.com/RocketChat/Rocket.Chat.ReactNative/assets/47038980/3b9d1fb4-a917-44d5-9de6-da5d7983c72f

#### After

https://github.com/RocketChat/Rocket.Chat.ReactNative/assets/47038980/a629e45f-cf1c-4d60-8a3e-f3bd0aed766b


## Screenshots


#### Before

<img width="320" alt="Screen Shot 2024-02-01 at 14 06 24" src="https://github.com/RocketChat/Rocket.Chat.ReactNative/assets/47038980/e55ad428-fe3d-4870-84ae-443656d924ba">


#### After

![Screen Shot 2024-02-01 at 21 56 25](https://github.com/RocketChat/Rocket.Chat.ReactNative/assets/47038980/ef658d5f-4d15-42f3-a708-198b38fa88f7)

## Types of changes




## Checklist



## Further comments

[TC-1093]

[TC-1093]: https://rocketchat.atlassian.net/browse/TC-1093?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
